server/middleware/doubleClickPreventionMiddleware.ts

```diff
@@ -0,0 +1,32 @@
+import type { RequestHandler } from 'express'
+import crypto from 'crypto'
+
+import { reseller } from 'googleapis/build/src/apis/reseller'
+import logger from '../../logger'
+import LockManager from '../data/lockManager'
+
+const LOCK_ID_KEY = 'lockId'
+
```

Comment:
I had implemented an unlock also but it added a load of complexity and the lock is at the page view level rather than locking an entire arrival so it didnt seem to warrant the complexity - a page refresh would generate a new key so allow another lock to be created